src/value.c

```diff
@@ -517,8 +562,66 @@ int RSValue_Cmp(const RSValue *v1, const RSValue *v2) {
   return cmp_strings(s1, s2, l1, l2);
 }
 
-int RSValue_Equal(const RSValue *v1, const RSValue *v2) {
-  return RSValue_Cmp(v1, v2) == 0;
+int RSValue_Equal(const RSValue *v1, const RSValue *v2, QueryError *qerr) {
+  assert(v1);
+  assert(v2);
+  v1 = RSValue_Dereference(v1);
+  v2 = RSValue_Dereference(v2);
+
+  if (v1->t == v2->t) {
```

Comment:
@rafie Why don't you just call RSValue_Cmp as before? seems like needless duplication